accumulator/utils.go

```diff
@@ -9,6 +9,85 @@ import (
 // verbose is a global const to get lots of printfs for debugging
 var verbose = false
 
+// ProofPositions returns the positions that are needed to prove that the targets exist.
+func ProofPositions(targets []uint64, numLeaves uint64, forestRows uint8) ([]uint64, []uint64) {
+	// the proofPositions needed without caching.
+	var proofPositions, computedPositions []uint64
+	for row := uint8(0); row < forestRows; row++ {
+		computedPositions = append(computedPositions, targets...)
+		if numLeaves&(1<<row) > 0 && len(targets) > 0 &&
```

Comment:
```numLeaves&(1<<row) > 0``` is checking to see if a root exists right? Comment or maybe even splitting this up to make something like ```rootExists := numLeaves&(1<<row) > 0``` could be better for readability.

Reply:
we could probably do this in a lot of places or maybe we create a function that checks for roots that we can call in various places, pretty sure the go compiler would inline simple functions like that.

Reply:
wasn't there some example very recently where Go didn't inline a really simple function? probably something i read from @kcalvinalvin

Reply:
it should inline all functions that meet these [criteria](https://github.com/golang/go/wiki/CompilerOptimizations#function-inlining).
we can also verify that it inlines them using `-gcflags=-m` when building.